### PR TITLE
updated DEVELOPER_GUIDE.md with changes for jenkins test

### LINKED ISSUE #1560: Update instructions for jenkins tests in DEVELOPER_GUIDE.md
### Is your feature request related to a problem? Please describe

We need to update the `developer_guide.md` with instructions on how to write jenkins test cases.

### Describe the solution you'd like

Update `developer_guide.md` with steps to write jenkins test cases for - 
- [x] jenkins job
- [x] jenkins libraries

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by abhinavGupta16**
Signed-off-by: Abhinav Gupta <abhng@amazon.com>

### Description
Updated `DEVELOPER_GUIDE.md` with steps for write tests for jenkins
 
### Issues Resolved
#1560 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -234,17 +234,54 @@ BUILD SUCCESSFUL in 7s
...
 #### Regression Tests
 
-Jenkins workflow regression tests typically output a .txt file into [tests/jenkins/jobs](tests/jenkins/jobs). For example, [TestHello.groovy](tests/jenkins/TestHello.groovy) executes [Hello_Jenkinsfile](tests/jenkins/jobs/Hello_Jenkinsfile) and outputs [Hello_Jenkinsfile.txt](tests/jenkins/jobs/Hello_Jenkinsfile.txt). If the job execution changes, the regression test will fail. To update the recorded .txt file run `./gradlew test -info -Ppipeline.stack.write=true` or update its value in [gradle.properties](gradle.properties).
+Jenkins workflow regression tests typically output a .txt file into [tests/jenkins/jobs](tests/jenkins/jobs).
+For example, [TestHello.groovy](tests/jenkins/TestHello.groovy) executes [Hello_Jenkinsfile](tests/jenkins/jobs/Hello_Jenkinsfile)
+and outputs [Hello_Jenkinsfile.txt](tests/jenkins/jobs/Hello_Jenkinsfile.txt). If the job execution changes, the regression test will fail.
+
+- To update the recorded .txt file run `./gradlew test -info -Ppipeline.stack.write=true` or update its value in [gradle.properties](gradle.properties).
+
+- To run a specific test case, run `./gradlew test -info -tests=TestCaseClassName`
```
Comment (zelinh) [inline DEVELOPER_GUIDE.md:243]: We could introduce the functionalities of different params here. 
e.g. `-info` will give the results in details. 
`-tests=TestCaseName` can be applied to run test on specific test class. 
`-Ppipeline.stack.write=true` will give the write access to update the `.txt` file

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -234,17 +234,54 @@ BUILD SUCCESSFUL in 7s
...
+
+#### Tests for jenkins job
+All jenkins jobs should have a test case associated with it.
```
Comment (zelinh) [inline DEVELOPER_GUIDE.md:246]: Each Jenkins job*

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -234,17 +234,54 @@ BUILD SUCCESSFUL in 7s
...
+  - Note: This will not include the variables set using function arguments
+- implement `void libFunctionName()`. This function will contain the name of function.
+- implement `void parameterInvariantsAssertions()`. This function will contain assertions verifying the type and 
+accepted values for the function parameters
+- implement `void expectedParametersMatcher()`. This function will match args called in the job to expected values from 
+the test
+
+##### Library Test Case
+
+All jenkins libraries should have a test case associated with it. Eg: [TestSignArtifacts](tests/jenkins/TestSignArtifacts.groovy) <br>
```
Comment (zelinh) [inline DEVELOPER_GUIDE.md:270]: ^^Each Jenkins library*

**Review by zelinh — COMMENTED** [review]
Thanks for drafting update for this guide!

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (abhinavGupta16) [reply DEVELOPER_GUIDE.md:243]: `./gradlew --help` should give info about all the option params

**Review by dblock — APPROVED** [review]

**Review by zelinh — APPROVED** [review]
Thanks!

**Pull request merged**